instrumentation/jmx-metrics/library/src/test/java/io/opentelemetry/instrumentation/jmx/engine/RuleParserTest.java

```diff
@@ -312,25 +359,104 @@ void testConf8() throws Exception {
     assertThat(mb1.getUnit()).isNull();
   }
 
+  private static final String CONF9 =
+      "---                                   # keep stupid spotlessJava at bay\n"
+          + "rules:\n"
+          + "  - bean: my-test:type=9\n"
+          + "    mapping:\n"
+          + "      jmxStateAttribute:\n"
+          + "        type: updowncounter\n"
+          + "        metric: state_metric\n"
+          + "        stateMapping:\n" // --> implies it's a state metric, value is either 0 or 1
+          + "          ok: STARTED\n" // as simple string
+          + "          failed: [STOPPED,FAILED]\n" // as array of strings
+          + "          degraded: '*'\n" // wildcard value for default
+          + "        metricAttribute:\n"
+          + "          state: statekey()\n" // --> only one state attribute allowed
+          + "";
```

Comment:
[for reviewer] this tests the new state metrics parsing and also the actual values captured.